tests/unit/test_rest_streaming.py

```diff
@@ -0,0 +1,211 @@
+            self._responses_bytes = _response_bytes
+            self._i = 0
+            self._random_split = random_split
+
+        def __next__(self):
+            if self._i == len(self._responses_bytes):
+                raise StopIteration
+            if self._random_split:
+                n = random.randint(1, len(self._responses_bytes[self._i :]))
+            else:
+                n = 1
+            x = self._responses_bytes[self._i : self._i + n]
+            self._i += n
+            return x.decode("utf-8")
+
+    def __init__(
+        self, responses: List[proto.Message], response_cls, random_split=False,
+    ):
+        super().__init__()
+        self._responses = responses
+        self._random_split = random_split
+        self._response_message_cls = response_cls
+
+    def _parse_responses(self, responses: List[proto.Message]) -> bytes:
+        # json.dumps returns a string surrounded with quotes that need to be stripped
+        # in order to be an actual JSON.
+        json_responses = [
+            self._response_message_cls.to_json(r).strip('"') for r in responses
+        ]
+        logging.info(f"Sending JSON stream: {json_responses}")
+        ret_val = "[{}]".format(",".join(json_responses))
+        return bytes(ret_val, "utf-8")
+
+    def close(self):
+        raise NotImplementedError()
+
+    def iter_content(self, *args, **kwargs):
+        return self._ResponseItr(
+            self._parse_responses(self._responses), random_split=self._random_split,
+        )
+
+
+@pytest.mark.parametrize("random_split", [False])
+def test_next_simple(random_split):
+    responses = [EchoResponse(content="hello world"), EchoResponse(content="yes")]
+    resp = ResponseMock(
+        responses=responses, random_split=random_split, response_cls=EchoResponse
+    )
+    itr = rest_streaming.ResponseIterator(resp, EchoResponse)
+    assert list(itr) == responses
+
+
+@pytest.mark.parametrize("random_split", [True, False])
+def test_next_nested(random_split):
+    responses = [
+        Song(title="some song", composer=Composer(given_name="some name")),
+        Song(title="another song", date_added=datetime.datetime(2021, 12, 17)),
+    ]
+    resp = ResponseMock(
+        responses=responses, random_split=random_split, response_cls=Song
+    )
+    itr = rest_streaming.ResponseIterator(resp, Song)
+    assert list(itr) == responses
+
+
+@pytest.mark.parametrize("random_split", [True, False])
+def test_next_stress(random_split):
+    n = 50
+    responses = [
+        Song(title="title_%d" % i, composer=Composer(given_name="name_%d" % i))
+        for i in range(n)
+    ]
+    resp = ResponseMock(
+        responses=responses, random_split=random_split, response_cls=Song
+    )
+    itr = rest_streaming.ResponseIterator(resp, Song)
+    assert list(itr) == responses
+
+
+@pytest.mark.parametrize("random_split", [True, False])
+def test_next_escaped_characters_in_string(random_split):
```

Comment:
Just to be sure, does this test fail when your string escape changes are not present?

Reply:
Yes, it does.

Reply:
Good, just a quirk of mine to disprove the null hypothesis.